src/rust/engine/fs/fs_util/src/main.rs

```diff
@@ -410,19 +480,10 @@ pub fn fingerprint_to_digest(
   digest
 }
 
-fn file_name_as_utf8(path: &Path) -> Result<String, String> {
-  match path.file_name() {
-    Some(name) => {
-      match name.to_str() {
-        Some(name_utf8) => Ok(name_utf8.to_string()),
-        None => Err(format!(
-          "{:?}'s file_name is not representable in UTF8",
-          path
-        )),
-      }
-    }
-    None => Err(format!("{:?} did not have a file_name", path)),
-  }
+fn osstring_as_utf8(path: &OsStr) -> Result<String, String> {
+  path.to_str().map(|e| e.to_string()).ok_or_else(|| {
```

Comment:
It looks like all callers have owned `OsString` values, so I think you can update them to pass the owned values and write this as:
```
fn osstring_as_utf8(path: OsString) -> Result<String, String> {
  path.into_string().or_else(|e| {
    Err(format!("{:?}'s file_name is not representable in UTF8", e))
  })
}
```
...to avoid copying.

Reply:
Done